go-controller/pkg/libovsdb/ops/model.go

```diff
@@ -493,7 +493,7 @@ func buildFailOnDuplicateOps(c client.Client, m model.Model) ([]ovsdb.Operation,
 func getAllUpdatableFields(model model.Model) []interface{} {
 	switch t := model.(type) {
 	case *nbdb.LogicalSwitchPort:
-		return []interface{}{&t.Addresses, &t.Type, &t.TagRequest, &t.Options, &t.PortSecurity}
+		return []interface{}{&t.Addresses, &t.Type, &t.TagRequest, &t.Options, &t.PortSecurity, &t.Enabled}
```

Comment:
We can have a new method that takes a fields argument as it is already done elsewhere (see `CreateOrUpdateLogicalSwitch` for example) which is what I originally suggested.

Other option is to implement a `getAllNonZeroFields` as a more sensible default than the current `getAllUpdatableFields` implementation. It does require a bit more work in validating this default is suitable for current uses which it should.

Reply:
I see what you mean. I remember I had issues with it, but let me play with it some more and update on this thread.

Reply:
oh I remember now. 
We can't do something like `UpdateLogicalSwitchPort` because it also transacts the change, but the we need to remember that the LSP was not created yet when reconciling a new pod - the ops that will create it will only transacted [later in the code](https://github.com/ovn-kubernetes/ovn-kubernetes/blob/9cf5a7707ca278b96fed631a8780fbf3ff7e32bb/go-controller/pkg/ovn/base_network_controller_secondary.go#L368).
I Also tried to just create the ops for changing the LSP.Enalbed to false/true and appending to the current ops, but that's not a valid operation as discussed with @jcaamano.

Reply:
Right, more or less since we are at the same transaction we have no way to make the Enabled field looks like an update since the LSP does not exist yet and for this PR having enabled as false at creation time is important.

Reply:
I don't understand. I mentioned `CreateOrUpdateLogicalSwitch` not `UpdateLogicalSwitchPort`. It's the same use case as `CreateOrUpdateLogicalSwitch`, why wouldn't it work?

Reply:
>I don't understand. I mentioned CreateOrUpdateLogicalSwitch not UpdateLogicalSwitchPort. It's the same use case as CreateOrUpdateLogicalSwitch, why wouldn't it work?

`CreateOrUpdateLogicalSwitch` creates the ops and immediately transacts them. we can't do that, because the in some cases (=on first pod creation reconcile) it would mean we are:
1. creating the regular lsp with ops1, but not transacting them
2. creating the small diff of lsp.Enabled to ops2 and transacting it (effectivly creating the lsp with wrong params)
3. transacting ops2

This scenario seems wrong to me, so I went to do all the lsp changes (including the lsp.Enabled) before the Ops are created, then transaction it the regular way..

Reply:
@jcaamano But anyways I decided to go for your second suggestion, implementing `getNonZeroFields`. Is this what you intended?

Reply:
If you don't want to transact, add a method that does not transact (i.e `CreateOrUpdateLogicalSwitchPortOps`, see `CreateOrUpdateMeterOps` as a reference).